### PR TITLE
tigerbeetle: 0.14.171 -> 0.14.175

### PR DESCRIPTION
**Pull request by DanielSidhion**
## Description of changes

The tigerbeetle folks stopped exposing the `-Doptimize` build flag in https://github.com/tigerbeetle/tigerbeetle/commit/94cacce42860599378901f941ab14ff2283249d7, which means the default Nixpkgs zig build hook makes the tigerbeetle build fail on version 0.14.174 (currently the latest).

This PR fixes the build by overriding the default zig build hook and omitting the `-Doptimize` flag if it exists.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ti/tigerbeetle/package.nix [hunk 1]
```diff
@@ -6,20 +6,27 @@
 , tigerbeetle
 , nix-update-script
 }:
+let
+  # Read [these comments](pkgs/development/compilers/zig/hook.nix#L12-L30) on the default Zig flags, and the associated links. tigerbeetle stopped exposing the `-Doptimize` build flag, so we can't use the default Nixpkgs zig hook as-is. tigerbeetle only exposes a boolean `-Drelease` flag which we'll add in the tigerbeetle derivation in this file.
```
Comment (a-n-n-a-l-e-e) [inline pkgs/by-name/ti/tigerbeetle/package.nix:10]: is this worthy of filing an issue (or patch) upstream to add back support for optimize?

**Review by ghost — COMMENTED** [review]
LGTM -- one comment inline.

File: pkgs/by-name/ti/tigerbeetle/package.nix (hunk 1 above)
Comment (a-n-n-a-l-e-e) [reply pkgs/by-name/ti/tigerbeetle/package.nix:10]: reading the PR comment I'd say no.
https://github.com/tigerbeetle/tigerbeetle/pull/1362

**Comment by a-n-n-a-l-e-e** [comment]
one thing i didn't realize about this package is they have weekly releases which aren't stable.
https://github.com/tigerbeetle/tigerbeetle#tigerbeetle which seems to limit its utility.

**Comment by a-n-n-a-l-e-e** [comment]
aarch64-darwin is [failing](https://github.com/NixOS/nixpkgs/pull/278460/checks?check_run_id=20154106516)

**Review by ghost — APPROVED** [review]
LGTM

**Comment by DanielSidhion** [comment]
@a-n-n-a-l-e-e thanks for the review! I don't have access to a mac system, which makes that build failure much harder to debug. Looking around at the zig source code for that error, I think I'm almost sure it's coming from https://github.com/ziglang/zig/blob/b80cad24840700ead20b59734916bf3d4d4ba87c/lib/std/zig/system/darwin.zig#L15-L19
which (if I understand correctly) will never succeed in Nix builds. Either we'll need more zig patches in nixpkgs, or zig has to fix that on their code.

For now, I removed the darwin support flag for this package, because it seems to be something I can't fix right now, and/or that would require me to go on a loop with ofborg until I find the magical words to get it to build (and I'm unwilling to go through this pain).

I'm apprehensive to open an issue with zig though, because I just don't have a mac system to provide any meaningful support in debugging and/or verifying any fix.

As for the weekly releases - they don't have any "stable" releases so far, but they're aiming to have their first stable release this year and are tracking the remaining work in https://github.com/tigerbeetle/tigerbeetle/issues/259. I'm building all the support in nixpkgs/nixos that I need for tigerbeetle already because I'm already using it in my own preproduction stack, and will continue to use it after it has its first stable release.

**Pull request merged**